### DISCUSSION TITLE
How do you update Filebrowser?

### DISCUSSION THREAD

**Discussion by GH0STDATA** [swizzin/swizzin#390] [Q&A + Help] [answered]
Sorry about posting that question in the issues section. I'll ask the question again here: I'm just wondering how I would go about properly updating the filebrowser application included with Swizzin. It's currently on 2.1.0 and the latest version is 2.6.2. If anyone could help here, that would be appreciated.

**Comment by liaralabs** [comment, marked as answer]
Unfortunately I don't use Filebrowser and it was a user-contribution so there may have been some things (like the upgrade script) that were overlooked during it's inclusion. I can look at creating an upgrade script, but I can't make promises as to when this will happen. However, if you're willing to test a few commands and report back, it could help speed up the inclusion of an upgrader.

Run these commands as your user (not root) and make sure you define the username variable with your own username
```
username={YOUR USERNAME}

mv /home/${username}/bin/filebrowser /home/${username}/bin/filebrowser.bak
wget -qO "/home/${username}/filebrowser.tar.gz" "$(curl -sNL https://api.github.com/repos/filebrowser/filebrowser/releases/latest | grep -Po 'ht(.*)linux-amd64(.*)gz')"
tar -xvzf "/home/${username}/filebrowser.tar.gz" --exclude LICENSE --exclude README.md -C "/home/${username}/bin"
rm -f "/home/${username}/filebrowser.tar.gz"
chmod 700 "/home/${username}/bin/filebrowser"
sudo systemctl restart filebrowser
```

This should grab the new binary and drop it in place while backing up the old binary (in case of troubles) and restart filebrowser with the new binary in place.

**Reply by GH0STDATA** [reply]
Nicely done, it worked without a problem. Thank you very much for taking the time to actually help me with this!

**Reply by liaralabs** [reply]
Thanks for testing! I've gone ahead and tossed this into an upgrade script, for the future you should be able to simply use 

```
box upgrade filebrowser
```
